default.nix

```diff
@@ -1,183 +1,6 @@
-
-    { stdenv
-    , runCommandNoCC
-    , lib
-    , ruby
-    , bundlerEnv
-    # Groups installed in the bundler environment.
-    , groups ? [ "default" ]
-    }:
-
-    let
-      # Using `fetchGit` helps reduce pollution.
-      # This is because instead of copying the whole directory, it copies the
-      # checked-out revision + any files tracked or staged.
-      src = builtins.fetchGit ./.;
-
-      # All unique groups referenced in `gemset.nix`; should map to all groups
-      # actually present in the Gemfile.
-      allGroups = lib.unique (builtins.concatLists (
-        lib.mapAttrsToList
-        (k: v: v.groups)
-        (import (src + "/gemset.nix"))
-      ));
-
-      # [ "default" "development" "test" ] - [ "default ] ⇒ [ "development" "test" ]
-      selectedGroups = lib.subtractLists groups allGroups;
-
-      gemfile =
-        if groups == null
-        then src + "/Gemfile"
-        else (runCommandNoCC "patched-gemfile" {} ''
-          cat ${src}/Gemfile > $out
-          cat >> $out <<EOF
-
-          # Mark unwanted groups as "optional" such that we don't need to rely
-          # on "BUNDLE_WITHOUT" listing all these groups in an additional layer
-          # of wrappers to make bundler happy with missing gems.
-          ${lib.concatMapStringsSep "\n" (
-            name: "group :${name}, optional: true do end"
-          ) selectedGroups}
-          EOF
-        '')
-      ;
-      gems = bundlerEnv {
-        name = "rails-nix-gems";
-        inherit ruby groups gemfile;
-        gemdir = src;
-      };
-    in
-    stdenv.mkDerivation {
-      pname = "rails-nix";
-      version = "0.0.0";
-
-      inherit src;
-
-      buildInputs = [
-        gems
-        gems.wrappedRuby
-      ];
-
-      installPhase = ''
-        cp -vr . $out
-
-        (
-        cd $out
-
-        # build the assets
-        bundle exec rails assets:precompile
-        
-        # Clean up any cache artifacts from assets precompilation.
-        rm -r tmp/cache
-        )
-
-        ${lib.optionalString (runtimeDirectory != null) ''
-        (
-        cd $out
-        # `tmp` is forced to be relative to `Rails.root`.
-        rm -vrf tmp
-        ln -vs "${runtimeDirectory}/tmp" tmp
-
-        # TODO Make presence of sqlite for production use configurable
-        # These databases could be configured differently if sqlite was desired.
-        ln -vs "${runtimeDirectory}/db/development.sqlite3" db/development.sqlite3
-        ln -vs "${runtimeDirectory}/db/production.sqlite3" db/production.sqlite3
-        ln -vs "${runtimeDirectory}/db/schema.rb" db/schema.rb
-        )
-        ''}
-      '';
-
-      passthru = {
-        inherit ruby;
-        inherit gems;
-        inherit (gems) wrappedRuby;
-      };
-    }
-
-  ) {
-    #
-    # If the build required, let's say, `ruby_2_6`, this could be injected like so:
-    #
-    # ```
-    # ruby = pkgs.ruby_2_6;
-    # ```
-    #
+((import ./fly-base.nix {}).fly.evalSpec) {
```

Comment:
What does `evalSpec` mean here?

Reply:
- https://github.com/fly-apps/nix-base/blob/e1888f046dda4029841b884694be4e757b125094/overlay.nix#L15
 - https://github.com/fly-apps/nix-base/blob/e1888f046dda4029841b884694be4e757b125094/support/eval-spec.nix

Using "spec" might have been too early... I didn't know what to name it, so I hung on the "app spec" stuff eagerly.

The important thing is that it's the entry point to the modules system